extensions/chghost-3.2.md

```diff
@@ -5,59 +5,87 @@ copyrights:
 
-This message represents that the user identified by nick!user@host has changed
-host to another value. The first parameter is the user of the client. The
-second parameter is the new host the client is using.
+The `CHGHOST` message is as follows:
 
-On irc daemons with support for changing the user portion of a client, the
-second form may appear:
+    :nick!old_user@old_host.local CHGHOST new_user new_host.local
 
-    :nick!user@host CHGHOST newuser host
+The `new_user` parameter represents the user's "username" or "ident" which may
+or may not have changed in the CHGHOST process.
 
-If specified, a client may also have their user and host changed at the same
-time:
+The `new_host.local` parameter represents the new hostname for the user which
+may or may not have changed in the CHGHOST process.
 
-    :nick!user@host CHGHOST newuser new.host.goes.here
+Servers that implement ident checking might choose to prefix a username with a tilde `~` character to indicate missing confirmation from an ident server. This MUST be considered part of the username and included in any CHGHOST messages where relevant.
 
-This second and third form should only be seen on IRC daemons that support
-changing the user field of a user.
+    :nick!~old_user@old_host.local CHGHOST ~new_user new_host.local
 
-In order to take full advantage of the CHGHOST message, clients must be modified
-to support it. The proper way to do so is this:
+## Fallback when the `chghost` capability has not been negotiated
 
-1) Enable the chghost capability at capability negotiation time during the
-   login handshake.
+When the capability is not enabled for other clients who share channels with or
+monitor the changed client, servers SHOULD send fallback messages to simulate the client
+reconnecting. This allows clients to keep their user state up to date. For
+shared channels, the simulated events SHOULD include appropriate `QUIT`, `JOIN`
+and `MODE` commands, to restore membership and user channel modes. For monitored
+clients, the events SHOULD include appropriate `RPL_MONOFFLINE` and
+`RPL_MONONLINE` numerics.
 
-2) Update the user and host portions of data structures and process channel
-   users as appropriate.
+    :nick!old_user@old_host.local QUIT :Changing hostname
+    :nick!new_user@new_host.local JOIN #ircv3
+    :ircd.local MODE #ircv3 +v :nick
 
 ## Examples
 
-In this example, `tim!~toolshed@backyard` gets their username changed to `b` and
+In this example, `tim!~toolshed@backyard` gets their username changed to `~b` and
 their hostname changed to `ckyard`:
 
-    :tim!~toolshed@backyard CHGHOST b ckyard
+    :tim!~toolshed@backyard CHGHOST ~b ckyard
 
-In this example, `tim!b@ckyard` gets their username changed to `~toolshed` and
+In this example, `tim!b@ckyard` gets their username changed to `toolshed` and
 their hostname changed to `backyard`:
 
-    :tim!b@ckyard CHGHOST ~toolshed backyard
+    :tim!b@ckyard CHGHOST toolshed backyard
 
 ## Errata
 
```

Comment:
The requirement for monitored clients seems to be new, but isn't mentioned here.